Rework Lua module searchers and loaders (#9511)
tarantool/tarantool: changelogs/unreleased/fix-lua-loader-error-hendling.md

## bugfix/lua

> * Fixed the misbehaviour with Lua error propagation when the error is raised
>   while loading the module via the Tarantool-specific loaders.

Reviewer: ```suggestion
* Fixed the incorrect propagation of Lua errors raised while loading
  the module via the Tarantool-specific loaders.
```

Author: Thanks for the suggestion! There was also a typo in the file name.

```diff
diff --git a/changelogs/unreleased/fix-lua-loader-error-handling.md b/changelogs/unreleased/fix-lua-loader-error-handling.md
new file mode 100644
index 0000000000..518f7268a7
--- /dev/null
+++ b/changelogs/unreleased/fix-lua-loader-error-handling.md
@@ -0,0 +1,4 @@
+## bugfix/lua
+
+* Fixed the incorrect propagation of Lua errors raised while loading
+  the module via the Tarantool-specific loaders.
diff --git a/changelogs/unreleased/fix-lua-loader-error-hendling.md b/changelogs/unreleased/fix-lua-loader-error-hendling.md
deleted file mode 100644
index 9ad041050c..0000000000
--- a/changelogs/unreleased/fix-lua-loader-error-hendling.md
+++ /dev/null
@@ -1,4 +0,0 @@
-## bugfix/lua
-
-* Fixed the misbehaviour with Lua error propagation when the error is raised
-  while loading the module via the Tarantool-specific loaders.
```
